### PR TITLE
[CI/Build] Remove stale xfail marks from min_tokens e2e tests

### PR DESCRIPTION
**Pull request by MSraye**
## Purpose

`tests/v1/e2e/general/test_min_tokens.py` carries six `pytest.mark.xfail(strict=False)` markers that are now stale:

- Four reference #21987 ("stop strings bypass `min_tokens`"), and their own `reason` text already states the issue was **fixed by #22014**.
- Two are speculative *"potential logits-processor bug"* markers for EOS handling.

All six cases pass on current `main`. Because the markers are `strict=False`, the resulting `XPASS` outcomes are silently swallowed, so these tests no longer guard against regressions — if `min_tokens` behavior regressed, the affected cases would simply flip back to `xfail` (green) instead of failing.

This PR removes the stale `xfail` markers so the tests act as real regression guards, and rewords the now-misleading *"known bug" / "reproduces the bug"* docstrings and assertion prefixes accordingly. **No test logic, parameters, or assertions are changed** (only marker removal + comment wording).

## Test Plan

Ran the affected file against a CPU build with `facebook/opt-125m`:

```
VLLM_ENABLE_V1_MULTIPROCESSING=0 VLLM_CPU_KVCACHE_SPACE=4 \
  python -m pytest tests/v1/e2e/general/test_min_tokens.py -v
```

Also ran `ruff==0.14.0 check` and `ruff format --check` on the file.

## Test Result

- Before: `7 passed, 6 xpassed`
- After: `13 passed, 0 xpassed`
- `ruff check` + `ruff format --check`: pass

A future regression in `min_tokens` handling will now fail CI instead of being masked by a stale `xfail`.

---

*This PR was prepared with the assistance of GitHub Copilot (AI) and reviewed/submitted by the author.*

### CODE REVIEW

**Comment by MSraye** [comment]
@claude review

**Comment by MSraye** [comment]
Hi maintainers ” this is my first PR to vLLM, so `pre-run-check` is blocked pending a `ready`/`verified` label (understood, that's expected for a first-time contributor).

A brief summary to help triage: this is a small test-hygiene change (1 file) in `tests/v1/e2e/general/test_min_tokens.py`. It removes six stale `xfail(strict=False)` markers ” four reference #21987, whose own `reason` text notes it was already fixed by #22014, plus two speculative EOS markers. All six cases pass on current `main`; because the marks are `strict=False`, the resulting XPASS outcomes are silently swallowed, so the tests no longer guard against regressions. Removing the marks turns them back into real regression guards; no test logic, parameters, or assertions change.

Verified locally (CPU backend, `facebook/opt-125m`): before `7 passed, 6 xpassed` vs after `13 passed, 0 xpassed`. `ruff==0.14.0` check + format pass, and DCO is signed.

Could a maintainer take a look and, if it looks reasonable, apply the `ready` label to trigger CI? Thanks very much for your time and for maintaining vLLM.
